pkg/kapp/resources/mod_field_copy.go

```diff
@@ -73,7 +74,7 @@ func (t FieldCopyMod) apply(obj interface{}, path Path, fullPath Path, srcs map[
 			}
 
 		case part.ArrayIndex != nil:
-			if isLast {
+			if isLast && part.RegexObj == nil {
```

Comment:
Question: Can both ArrayIndex and RegexObj be non-nil at the same time?

Reply:
No so as of now we will be having one nil that's how we are unmarshalling and as we have regex from ArrayIndex I can remove this condition as well.